crates/flipperzero/src/gui/mod.rs

```diff
@@ -1,3 +1,230 @@
 //! GUI service.
 
 pub mod canvas;
+pub mod icon;
+pub mod icon_animation;
+pub mod view;
+pub mod view_port;
+pub mod xbm;
+
+use crate::{
+    gui::{
+        canvas::CanvasView,
+        view_port::{ViewPort, ViewPortCallbacks},
+    },
+    input::InputEvent,
+};
+use core::{
+    ffi::c_char,
+    fmt::Debug,
+    ops::{Deref, DerefMut},
+};
+use flipperzero_sys::{
+    self as sys, furi::UnsafeRecord, Canvas as SysCanvas, Gui as SysGui, GuiLayer as SysGuiLayer,
+};
+
+/// System Gui wrapper.
+pub struct Gui {
+    raw: UnsafeRecord<SysGui>,
+}
+
+impl Gui {
+    /// Furi record corresponding to GUI.
+    pub const RECORD: *const c_char = sys::c_string!("gui");
+
+    pub fn new() -> Self {
+        // SAFETY: `RECORD` is a constant
+        let gui = unsafe { UnsafeRecord::open(Self::RECORD) };
+
+        Self { raw: gui }
+    }
+
+    pub fn as_raw(&self) -> *mut SysGui {
+        self.raw.as_raw()
+    }
+
+    pub fn add_view_port<VPC: ViewPortCallbacks>(
+        &mut self,
+        view_port: ViewPort<VPC>,
+        layer: GuiLayer,
+    ) -> GuiViewPort<'_, VPC> {
+        let raw = self.as_raw();
+        let view_port_ptr = view_port.as_raw();
+        let layer = layer.into();
+
+        // SAFETY: all pointers are valid and `view_port` outlives this `Gui`
+        unsafe { sys::gui_add_view_port(raw, view_port_ptr, layer) };
+
+        GuiViewPort {
+            parent: self,
+            view_port,
+        }
+    }
+
+    pub fn get_frame_buffer_size(&self) -> usize {
+        let raw = self.as_raw();
+        // SAFETY: `raw` is always a valid pointer
+        unsafe { sys::gui_get_framebuffer_size(raw) }
+    }
+
+    pub fn set_lockdown(&self, lockdown: bool) {
+        let raw = self.raw.as_raw();
+        // SAFETY: `raw` is always a valid pointer
+        unsafe { sys::gui_set_lockdown(raw, lockdown) }
+    }
+
+    // TODO: separate `GuiCanvas` (locking the parent)
```

Comment:
Has this been renamed?
```suggestion
    // TODO: separate `ExclusiveCanvas` (locking the parent)
```